src/Workspaces/Core/Portable/Shared/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -168,29 +168,39 @@ public static IMethodSymbol RenameParameters(this IMethodSymbol method, IList<st
             return updatedMethod.RenameParameters(parameterNames);
         }
 
-        public static IMethodSymbol RemoveAttributeFromParametersAndReturnType(
-            this IMethodSymbol method, INamedTypeSymbol attributeType,
+        public static IMethodSymbol RemoveInaccessibleAttributesAndAttributesOfType(
+            this IMethodSymbol method, ISymbol accessibleWithin, INamedTypeSymbol removeAttributeType,
             IList<SyntaxNode> statements = null, IList<SyntaxNode> handlesExpressions = null)
         {
-            if (attributeType == null)
+            if (removeAttributeType == null)
```

Comment:
Is it possible for removeAttributeType to be null and there to also be inaccessible attributes that should be removed? If so, the inaccessible attributes won't be removed here. If not, consider adding a comment explaining why that situation cannot arise.